app/models/location/suggestions.rb

```diff
@@ -0,0 +1,281 @@
+# frozen_string_literal: true
+
+# Return array of suggested Locations for a given user input (name or lat/lon)
+class Location < AbstractModel
+  # SQL string to calculate db grid size in lat/lon squares
+  # This is the db counterpart of pseudoarea, differing by rounding error
+  GRID_SQUARES = "IF(east >= west, east - west, 360 + east - west) " \
+                 "* (north - south)"
+
+  # max suggested size in lat/lon "squares"
+  # suggestions should not include a Location with a larger pseudoarea
+  MAX_SUGGESTED_SIZE = 6800
```

Comment:
#### What's a reasonable number to use here?
I have no idea. 6800 is huge.
There are now 4 locations larger than this: North America, Russia, Earth, Latin America. (There used to be 57, but I fixed the rest yesterday.)